### DISCUSSION TITLE
Formatted output on event notifications

### DISCUSSION THREAD

**Discussion by jonsberndt** [JSBSim-Team/jsbsim#978] [Ideas]
A thought came to mind recently when I was looking at the output for a scripted JSBSim run that included this:

```
Time Notify (Event 0) executed at time: 10800.005000
    Height AGL (ft):          = 1322438.953873
    Altitude (Geod, ft):      = 1322438.953873
    Latitude (Geod, deg):     = -16.380122
    Vehicle Radius (ft):      = 22242546.477401
    Inertial Vel Mag (ft/s):  = 25165.743754
    Body X Velocity (ft/s):   = 13005.916406
    Simulation Frame:         = 2160001.000000
    Density:                  = 0.000000
    Angular momentum (ft^2/s) = 559749936198.419312
    Inclination (deg)         = 51.551024
    Right ascension (deg)     = -127.585985
    Argument of perigee (deg) = 33.878903
    Period (s)                = 5561.311971
    Eccentricity              = 0.001243
    Apoapsis radius (ft)      = 22286174.389210
    Periapsis radius (ft)     = 22230825.123889
    True anomaly (deg)        = 305.143610
```

I'd like to have more control over the notification outputs. The precision in some of the items above is excessive. The event in the script looks like this:

```xml
    <event name="Time Notify" persistent="true">
      <description>Output message at 10 minute intervals</description>
      <notify>
        <property caption="Height AGL (ft):         "> position/h-agl-ft               </property>
        <property caption="Altitude (Geod, ft):     "> position/geod-alt-ft            </property>
        ...
        ...
        <property caption="Periapsis radius (ft)    "> orbital/periapsis-radius-ft     </property>
        <property caption="True anomaly (deg)       "> orbital/true-anomaly-deg        </property>
      </notify>
      <condition> simulation/sim-time-sec >= simulation/notify-time-trigger </condition>
      <set name="simulation/notify-time-trigger" value="600" type="FG_DELTA"/>
    </event>
```

I propose adding a "format" attribute in the \<property\> element:

```xml
  <property caption="Periapsis radius (ft)    " format="%8.2f"> orbital/periapsis-radius-ft     </property>
```

I looked at the code and I don't think it would be too difficult. (And, I am intending to implement this upon concurrence that it's not a bad idea.)

Thoughts?

**Comment by seanmcleod** [comment]
I thought there was already a way to specify the precision for property output, but doing a search to refresh my memory it looks like it's currently only available for output sockets - https://github.com/JSBSim-Team/jsbsim/pull/579, and also not on a per-property basis.

**Reply by bcoconni** [reply]
Yes and I would suggest using the same XML syntax for both use cases for consistency. For backward compatibility as well as simplicity (not everyone is familiar with C format syntax "%8.2f") I'd suggest sticking with the syntax that was implemented by @seanmcleod in #579.

**Comment by jonsberndt** [comment]
Yeah, so this is one place where I feel it would be worthwhile, but there could very well be other places too.
